src/power_grid_model/core/power_grid_model.py

```diff
@@ -459,7 +482,7 @@ def calculate_power_flow(
             error_tolerance=error_tolerance,
             max_iterations=max_iterations,
             calculation_method=calculation_method,
-            update_data=update_data,
+            update_data=(_map_to_componenttypes(update_data) if update_data is not None else None),
```

Comment:
i'm actually a little bit suprised that the linter doesn't complain about the import of functions prefixed `_`

Reply:
@mgovers I am not surprised, I am happy that it doesn't!